File: apps/backend/internal/modules/features/support/domain/support.go
package domain

import "time"

const (
	StatusPending    = "PENDING"
	StatusInProgress = "IN_PROGRESS"
	StatusResolved   = "RESOLVED"
	StatusRejected   = "REJECTED"
)

// SupportMessage mirrors the support_messages table. It intentionally has no
// reply/response text column — "menanggapi" (responding) only tracks a
// status transition, not a stored written reply.
type SupportMessage struct {
	ID         string
	TenantID   *string
	Email      string
	Phone      string
	Name       string
	Title      string
	Category   string
	Chronology string
	Status     string
	UserID     *string
	CreatedAt  time.Time
	UpdatedAt  time.Time
}

File: apps/sitivent-frontend/src/interfaces/features/support.ts
export interface SupportMessage {
  id: string;
  email: string;
  phone: string;
  name: string;
  title: string;
  category: string;
  chronology: string;
  status: string; // "PENDING" | "RESOLVED"
  userId: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateSupportMessageInput {
  email: string;
  phone: string;
  name: string;
  title: string;
  category: string;
  chronology: string;
}

export interface SupportMessagesResponse {
  success: boolean;
  data?: SupportMessage[];
  meta?: {
    total: number;
    page: number;
    lastPage: number;
  };
  error?: string;
}

export interface SupportMessageMutationResponse {
  success: boolean;
  data?: SupportMessage;
  error?: string;
}
